shell/src/main/java/alluxio/cli/bundler/command/CollectMetricsCommand.java

```diff
@@ -99,38 +101,57 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
   }
 
   private void masterMetrics(StringWriter outputBuffer, int i) throws IOException {
+    List<InetSocketAddress> masterAddresses =
+        ConfigurationUtils.getMasterRpcAddresses(mFsContext.getClusterConf());
+    boolean masterHostsDistinct =
+        masterAddresses.stream().map(InetSocketAddress::getHostName).distinct().count()
+            == masterAddresses.size();
+
     // Generate URL from config properties
-    String masterAddr;
+    InetSocketAddress primaryMasterAddr;
     try {
-      masterAddr = mFsContext.getMasterAddress().getHostName();
+      primaryMasterAddr = mFsContext.getMasterAddress();
     } catch (UnavailableException e) {
       String noMasterMsg = "No Alluxio master available. Skip metrics collection.";
```

Comment:
if there's no primary in the cluster, would it still make sense to try all masters? Maybe there's an issue with ratis and election is slow, but you can still find the metrics?